### PR TITLE
feat: config-driven repository factory with @o3co/auth-provider-repositories package

### LINKED ISSUE #10: refactor: create @o3co/auth-provider-standalone-core package with repository factory
## Summary

Create a `packages/standalone-core` package (`@o3co/auth-provider-standalone-core`) that provides:

1. **Config-driven Repository Factory** — `config.clients.*.type` determines which implementation to use
2. **Production Repository Implementations** — `HttpUserRepository`, `RedisCodeRepository` (moved from `templates/standalone/`)
3. **YAML client loading** — type `"yaml"` → `InMemoryClientRepository` + `loadYamlMap`

## Motivation

Currently `templates/standalone/src/app.mts` has inconsistent repository construction:
- `HttpUserRepository` and `RedisCodeRepository` take config objects directly
- `InMemoryClientRepository` requires manual YAML loading with `loadYamlMap`

A config-driven factory would unify this:

```typescript
import { createClientRepository, createUserRepository, createCodeRepository } from "@o3co/auth-provider-standalone-core";

const clientRepository = createClientRepository(config.clients.client);  // type: "yaml"
const userRepository = createUserRepository(config.clients.user);        // type: "http"
const codeRepository = await createCodeRepository(config.clients.code);  // type: "redis"
```

## Benefits

- Scaffolded projects depend on `standalone-core` via npm → updates propagate via `npm update`
- Config type change = implementation change (pluggable)
- `templates/standalone/` becomes very thin (just DI wiring)
- Resolves the template update propagation problem from the monorepo design

## Package structure

```
packages/
├── core/                    # Interfaces + InMemory implementations (existing)
└── standalone-core/         # Factory + Http/Redis/Yaml implementations (new)
    ├── src/
    │   ├── factory.mts      # createClientRepository, createUserRepository, createCodeRepository
    │   ├── HttpUserRepository.mts
    │   ├── RedisCodeRepository.mts
    │   └── index.mts
    └── package.json         # @o3co/auth-provider-standalone-core
```

## Context

Identified during monorepo restructure PR (#8). The YamlLoader helper and InMemory rename were done in #8, this issue covers the next step: factory pattern + package extraction.

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

- Add generic `RepositoryFactory<T>` to core with `register`/`create` API for config-driven repository construction
- Add `createDefaultFactories()` with InMemory defaults (yaml client, yaml user, memory code) — core works standalone for dev/eval
- Create `@o3co/auth-provider-repositories` package with `HttpUserRepository`, `RedisCodeRepository`, and `registerBuiltinRepositories()`
- Standalone template simplified to factory-based construction (3 lines)
- Scaffolder handles per-package version resolution via `versions.json`

## Packages

| Package | Changes |
|---------|---------|
| `@o3co/auth-provider-core` | +`RepositoryFactory<T>`, +`createDefaultFactories()`, config schema `type` fields |
| `@o3co/auth-provider-repositories` | New package: Http/Redis implementations + `registerBuiltinRepositories()` |
| `templates/standalone` | Simplified to factory pattern, repositories removed |
| `create-app` | Per-package version resolution |

## Test plan

- [x] Core: 122 tests passing (+ defaultExpiresIn validation)
- [x] Repositories: 21 tests passing
- [x] Create-app: 8 tests passing (scaffolder logic + version resolution)
- [x] Lint clean
- [x] Build passes for all packages

Closes #10

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/repositories/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		"test": "vitest run"
+	},
+	"imports": {
+		"#/*": "./src/*"
+	},
+	"dependencies": {
+		"redis": "^5.10.0"
+	},
+	"peerDependencies": {
+		"@o3co/auth-provider-core": "workspace:*"
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/repositories/package.json]: Fixed — peerDependencies now uses `^0.0.0` semver range. `workspace:*` kept only in devDependencies for monorepo resolution. Commit 1045339.

File: packages/core/src/repositories/RepositoryFactory.mts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	codeFactory: RepositoryFactory<CodeRepository>;
+} => {
+	const clientFactory = new RepositoryFactory<ClientRepository>("client");
+	clientFactory.register("yaml", (config) => {
+		return new InMemoryClientRepository(loadYamlMap(config.path as string, ClientEntrySchema));
+	});
+
+	const userFactory = new RepositoryFactory<UserRepository>("user");
+	userFactory.register("yaml", (config) => {
+		return new InMemoryUserRepository(loadYamlMap(config.path as string, UserEntrySchema));
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/core/src/repositories/RepositoryFactory.mts]: Fixed — yaml builders now validate `typeof config.path === 'string'` and throw descriptive error. Commit 1045339.

File: packages/core/src/config/application.schema.mts [hunk 3]
```diff
@@ -80,15 +80,18 @@ export const AppConfigSchema = z.object({
...
+			type: z.string().default("yaml"),
 			path: z.string().default("./config/clients.yaml"),
 		}),
 		user: z.object({
-			baseURL: z.string(),
+			type: z.string().default("yaml"),
+			baseURL: z.string().optional(),
+			path: z.string().optional(),
 			timeout: z.coerce.number().default(5000),
 		}),
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/core/src/config/application.schema.mts:92]: Fixed — `clients.user.path` now defaults to `./config/users.yaml`, matching the yaml type default. Config is valid out of the box for dev/eval. Commit 1045339.

File: create-app/src/index.mts [hunk 4]
```diff
@@ -53,9 +52,16 @@ export const scaffold = (targetDir: string, projectName: string): void => {
...
+	const versions = getPackageVersions();
+	for (const section of ["dependencies", "devDependencies", "peerDependencies"] as const) {
+		const deps = pkg[section];
+		if (!deps) continue;
+		for (const [name, version] of Object.entries(deps)) {
+			if (version === "workspace:*") {
+				deps[name] = `^${versions[name] ?? "0.0.0"}`;
+			}
+		}
 	}
```
[the remark this answers is not in the document]
Comment (y1o1) [reply create-app/src/index.mts:70]: Fixed — scaffolder now throws if `versions.json` is missing an entry for a workspace dependency, instead of falling back to `^0.0.0`. Commit 1045339.

File: packages/core/src/config/application.schema.mts [hunk 5]
```diff
@@ -80,15 +80,19 @@ export const AppConfigSchema = z.object({
...
+			authenticateByTokenUrl: z.string().optional(),
+			path: z.string().default("./config/users.yaml"),
 			timeout: z.coerce.number().default(5000),
 		}),
 		code: z.object({
-			endpointUri: z.string(),
+			type: z.string().default("memory"),
+			endpointUri: z.string().optional(),
 			password: z.string().optional(),
 			defaultExpiresIn: z.coerce.number().default(600),
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/core/src/config/application.schema.mts:97]: By design — schema fields are optional because they're type-dependent (yaml doesn't need endpointUri, redis doesn't need path). Builder-level runtime validation catches missing fields with clear errors. Commit 1045339 (builder validation) + b789b1a (config defaults).

File: packages/core/src/repositories/RepositoryFactory.mts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+	userFactory.register("yaml", (config) => {
+		if (typeof config.path !== "string") {
+			throw new Error('YAML user repository requires "path" in config');
+		}
+		return new InMemoryUserRepository(loadYamlMap(config.path, UserEntrySchema));
+	});
+
+	const codeFactory = new RepositoryFactory<CodeRepository>("code");
+	codeFactory.register("memory", (config) => {
+		return new InMemoryCodeRepository(config as { defaultExpiresIn?: number });
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/core/src/repositories/RepositoryFactory.mts]: Fixed — memory builder now coerces via `Number()` and validates for NaN. Commit 9c2bd0b.

File: packages/repositories/src/__tests__/registerBuiltinRepositories.test.mts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+describe("registerBuiltinRepositories", () => {
+	it("registers 'http' type in userFactory", async () => {
+		const userFactory = new RepositoryFactory<UserRepository>("user");
+		const codeFactory = new RepositoryFactory<CodeRepository>("code");
+
+		registerBuiltinRepositories({ userFactory, codeFactory });
+
+		await expect(userFactory.create({ type: "unknown" })).rejects.toThrow(/http/);
+	});
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/repositories/src/__tests__/registerBuiltinRepositories.test.mts:56]: Fixed — added positive-path test for http builder with msw (creates HttpUserRepository, exercises authenticate + authenticateByToken). Also added negative tests for missing authenticateUrl and endpointUri. Commit 9c2bd0b.

File: Makefile [hunk 8]
```diff
@@ -18,19 +16,8 @@ install:
 
 .PHONY: clean
 clean:
-	rm -rf packages/core/dist templates/standalone/dist create-app/dist
+	rm -rf packages/core/dist packages/repositories/dist templates/standalone/dist create-app/dist
 
 .PHONY: test
 test:
 	pnpm -r run test
```
[the remark this answers is not in the document]
Comment (y1o1) [reply Makefile:23]: Fixed — README updated to point to templates/standalone/Dockerfile for scaffolded projects. Commit 6375649.

File: create-app/src/index.mts [hunk 9]
```diff
@@ -87,18 +98,9 @@ export const main = (): void => {
...
-			`You can still use the scaffolded project. To finish setup, run 'pnpm install' inside '${projectName}'.`,
-		);
-	}
-
 	console.log(`\nDone! Created ${projectName} at ${targetDir}`);
 	console.log(`\nNext steps:`);
 	console.log(`  cd ${projectName}`);
-	console.log(`  pnpm run debug`);
+	console.log("  npm install");
+	console.log("  npm run debug");
```
[the remark this answers is not in the document]
Comment (y1o1) [reply create-app/src/index.mts:105]: Fixed — error message is now tool-agnostic. Commit 6375649.

File: templates/standalone/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,44 @@
+#############################################
+FROM node:25-alpine AS base
+
+RUN npm install -g corepack --force
+
+ENV HOME=/home/node
+ENV NODE_ENV=production
+
```
[the remark this answers is not in the document]
Comment (y1o1) [reply templates/standalone/Dockerfile:7]: Fixed — NODE_ENV=production moved to runtime stage only. Builder installs all deps including devDependencies. Commit 78ea58f.

File: templates/standalone/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+WORKDIR ${HOME}
+
+ADD package.json .npmrc* ./
```
[the remark this answers is not in the document]
Comment (y1o1) [reply templates/standalone/Dockerfile]: Fixed — removed .npmrc* glob, only copies package.json. Commit 78ea58f.

File: packages/core/src/repositories/RepositoryFactory.mts [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+	userFactory: RepositoryFactory<UserRepository>;
+	codeFactory: RepositoryFactory<CodeRepository>;
+} => {
+	const clientFactory = new RepositoryFactory<ClientRepository>("client");
+	clientFactory.register("yaml", (config) => {
+		if (typeof config.path !== "string") {
+			throw new Error('YAML client repository requires "path" in config');
+		}
+		return new InMemoryClientRepository(loadYamlMap(config.path, ClientEntrySchema));
+	});
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/core/src/repositories/RepositoryFactory.mts]: Fixed — registered "static" as backward-compat alias for "yaml" in both clientFactory and userFactory. Commit 78ea58f.

File: packages/core/src/repositories/RepositoryFactory.mts [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+	userFactory.register("yaml", yamlUserBuilder);
+	userFactory.register("static", yamlUserBuilder); // backward compat alias
+
+	const codeFactory = new RepositoryFactory<CodeRepository>("code");
+	codeFactory.register("memory", (config) => {
+		const defaultExpiresIn =
+			config.defaultExpiresIn != null ? Number(config.defaultExpiresIn) : undefined;
+		if (defaultExpiresIn !== undefined && Number.isNaN(defaultExpiresIn)) {
+			throw new Error('"defaultExpiresIn" must be a number');
+		}
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/core/src/repositories/RepositoryFactory.mts:84]: Fixed in b7658e6. Now validates with `Number.isFinite()` and `> 0`, rejecting Infinity, negative, and zero values. Added 4 test cases for the edge cases.

File: create-app/src/__tests__/index.test.mts [hunk 14]
```diff
@@ -33,13 +33,24 @@ describe("scaffold", () => {
 		expect(pkg.name).toBe("my-auth");
 	});
 
-	it("replaces workspace:* core dependency with caret version", () => {
+	it("replaces all workspace:* dependencies with caret versions", () => {
 		const targetDir = join(tempDir, "my-auth");
 		scaffold(targetDir, "my-auth");
 
 		const pkg = JSON.parse(readFileSync(join(targetDir, "package.json"), "utf-8"));
```
[the remark this answers is not in the document]
Comment (y1o1) [reply create-app/src/__tests__/index.test.mts:40]: Updated PR description to accurately reflect test coverage. create-app tests cover scaffolder logic and version resolution (no E2E template validation in this PR).

File: packages/core/src/federations/__tests__/google.test.mts [hunk 15]
```diff
@@ -48,9 +48,14 @@ const baseConfig = {
...
-		user: { baseURL: "http://localhost:3001", timeout: 5000 },
-		code: { endpointUri: "http://localhost:3002", defaultExpiresIn: 600 },
+		client: { type: "yaml", path: "./config/clients.yaml" },
+		user: {
+			type: "http",
+			authenticateUrl: "http://localhost:3001/user/authenticate",
+			authenticateByTokenUrl: "http://localhost:3001/user/authenticate/token",
+			timeout: 5000,
+		},
+		code: { type: "redis", endpointUri: "http://localhost:3002", defaultExpiresIn: 600 },
```
[the remark this answers is not in the document]
Comment (y1o1) [reply packages/core/src/federations/__tests__/google.test.mts]: Fixed in b7658e6. Changed to `redis://localhost:6379`.
